pyproject.toml

```diff
@@ -1,6 +1,77 @@
 [build-system]
-build-backend = "setuptools.build_meta:__legacy__"
-requires = [
+build-backend = "setuptools.build_meta"
+requires = [ "setuptools>=61.2" ]
+
+[project]
+name = "cpplint"
+version = "2.0.0"
+description = "Automated checker to ensure C++ files follow Google's style guide"
+readme = "README.rst"
+keywords = [ "c++", "lint", "python" ]
+license = { text = "BSD-3-Clause" }
+maintainers = [ { name = "cpplint Developers" } ]
+requires-python = ">=3.8"
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Environment :: Console",
+  "Intended Audience :: End Users/Desktop",
+  "License :: Freely Distributable",
+  "License :: OSI Approved :: BSD License",
+  "Natural Language :: English",
+  "Programming Language :: C++",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3.13",
+  "Topic :: Software Development :: Quality Assurance",
+]
+dependencies = [  ]
+
+optional-dependencies.dev = [
+  "flake8>=4.0.1",
+  "parameterized",
+  "pylint>=2.11",
+  "pytest",
+  "pytest-cov",
+  "pytest-timeout",
   "setuptools",
-  "wheel",
+  "testfixtures",
+  "tox<5",
+]
+optional-dependencies.test = [
+  "parameterized",
+  "pytest",
+  "pytest-cov",
+  "pytest-timeout",
+  "testfixtures",
+  "tox<5",
+]
+optional-dependencies.testing = [
+  "parameterized",
+  "pytest",
+  "pytest-cov",
+  "pytest-timeout",
+  "testfixtures",
+  "tox<5",
 ]
```

Comment:
Rebasing rn, is there a reason the text extra was duplicated lol

Reply:
.test, .testing, and .dev all existed in `setup.py` so I replicated them in `pyproject.toml` to be sure I did not break anything.

Once we merge the open pre-commit PRs the only tools we will need in `.dev` will be pytest-related.  At that point, we can drop .test and .testing and just have dev.

The other issue is that the version is hard coded in both pyproject.toml and in Python code.  That should be fixed by making version dynamic in pyproject.toml and read out of the Python code.